src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,239 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const Script = @import("lib.zig").Script;
+const StackError = @import("stack.zig").StackError;
+pub const engine = @import("engine.zig");
+const Opcode = @import("./opcodes/constant.zig").Opcode;
+const isUnnamedPushNDataOpcode = @import("opcodes/constant.zig").isUnnamedPushNDataOpcode;
+const ScriptNum = @import("lib.zig").ScriptNum;
+const asBool = Script.asBool;
+const asInt = Script.asInt;
+const testing = std.testing;
+
+/// Maximum script length in bytes
+const MAX_SCRIPT_SIZE = 10000;
+
+/// ScriptBuilder Errors
+const ScriptBuilderError = error{ ScriptTooLong, ScriptTooShort };
+
+///ScriptBuilder is a library to generate easier scripts, useful for faster testing
+pub const ScriptBuilder = struct {
+    /// Dynamic array holding the opcodes
+    script: std.ArrayList(u8),
+
+    ///Memory Allocator
+    allocator: Allocator,
+
+    /// Initialize a new ScriptBuilder
+    ///
+    /// # Arguments
+    /// - `allocator`: Memory allocator for managing engine resources
+    ///
+    /// # Returns
+    /// - `!ScriptBuilder`: use with `try`
+    pub fn new(allocator: Allocator, capacity: usize) !ScriptBuilder {
+        return ScriptBuilder{
+            .script = try std.ArrayList(u8).initCapacity(allocator, capacity),
+            .allocator = allocator,
+        };
+    }
+
+    ///Push an OPcode to the ScriptBuilder
+    pub fn addOpcode(self: *ScriptBuilder, op: Opcode) !*ScriptBuilder {
+        if (self.script.items.len >= MAX_SCRIPT_SIZE) {
+            return ScriptBuilderError.ScriptTooLong;
+        }
+        try self.script.append(op.toBytes());
+        return self;
+    }
+
+    ///Push an Int to the SciptBuilder
+    pub fn addInt(self: *ScriptBuilder, num: i32) !*ScriptBuilder {
```

Comment:
Actually I think you make is so that addInt is just

```zig
pub fn addInt(self: *ScriptBuilder, num: i32) !*ScriptBuilder {
 return self.addBytes(std.mem.asBytes(std.mem.nativeToLittle(i32, num)));
}
```